app/controllers/requests_controller.rb

```diff
@@ -23,6 +24,12 @@ def current_request
 
   protected
 
+  def check_if_proxy_sponsor
+    return if params[:proxy].present? || !current_user.sponsor?
```

Comment:
Can you help me replicate this? I tried a simple page with just a library id and just name / email and it worked for me.